src/vs/workbench/contrib/terminal/browser/terminalInstanceService.ts

```diff
@@ -85,7 +85,10 @@ export class TerminalInstanceService extends Disposable implements ITerminalInst
 	}
 
 	async getBackend(remoteAuthority?: string): Promise<ITerminalBackend | undefined> {
-		await this._lifecycleService.when(LifecyclePhase.Restored);
+		if (remoteAuthority) {
```

Comment:
The terminal in question has no remoteAuthority, but this removes the await for all local terminals, so maybe you have a lower impact way to do this.

Reply:
I was thinking about ways to make this a little more solid to prevent future problems like moving the lifecycle page into the backend, but we will always have a possible breakage from the workbench contribution phase needing to be defined separately.

How about we instead do this change:

```ts
// Return existing backends immediately
let backend = Registry.as<ITerminalBackendRegistry>(TerminalExtensions.Backend).getTerminalBackend(remoteAuthority);
if (!backend) { 
  // Ensure all backends are initialized and try again
  await this._lifecycleService.when(LifecyclePhase.Restored);
  backend = Registry.as<ITerminalBackendRegistry>(TerminalExtensions.Backend).getTerminalBackend(remoteAuthority);
}
return backend
```

I think this makes the backend have a single source of truth for its init lifecycle phase.